gri-is/portal
# Enhancement/frontend config/448

Closes #448 and handles related issue of frontend configuration for different environments.

I am using [gulp] (http://gulpjs.com/) and [gulp-ng-config](https://github.com/ajwhite/gulp-ng-config)
to generate an Angular module called `app.env.config` with configuration constants for host, port, etc which are based on `.json` files.

The `.json` files used to generate angular config settings are stored, respectively, in `config/local`, `config/develop`, and `config/prod`.

`config/develop` and `config/prod` are **not** tracked in version control, so on those respective boxes we can add configuration files there and they will not be overwritten when we pull down new source.

-----------

**First-time Test instructions**

1. Pull down this branch onto the server.
2. Run `npm install` or `sudo npm install` to install gulp & related packages.
3. From the portal root dir, run `gulp clean:config` to delete the default `app.env.config.js` file which is in `src/client/app/`.
4. Edit the appropriate `app.env.config.json` file in `config/` for your environment.
5. From portal root dir, run `gulp config:develop` if you are in development, `gulp config:prod` if you are in production, etc.
6. Start up the app; the app should now work. 

**Build Instructions when pulling down new code**
1. Pull down new code.
2. From portal root dir, run `gulp clean:config`
3. From portal root dir, run `gulp config:<my-environment-name-here>`

*(we can chain these into a `gulp build` command if desired or use a shell script, etc)*

@joshuago78 
I've left the default `src/client/app/app.env.config.js` for local dev environment in version control as a convenience for local development.We can add it to the `.gitignore`  -- but then devs will have to run `gulp config:local` every time they pull down code on to their machine. Let me know if you'd like me to remove it. 

------------
**PR Review**

- [x] @joshuago78 
- [ ] @jfranco-gri
